src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -263,6 +263,58 @@ private IList<Tracepoint> GetTracepoints(IDebugBreakpointEvent2 debugEvent)
             return tracepoints;
         }
 
+        /// <summary>
+        /// Converts a string expression into a IDebugProperty2 with given frame and evaluation flags.
+        /// </summary>
+        /// <param name="expression">Expression to evaluate</param>
+        /// <param name="frame">Frame to execute the expression on</param>
+        /// <param name="additionalEvalFlags">Additional evalFlags besides the flags normally used for vs watch window.</param>
+        /// <param name="additionalDapEvalFlags">Additional flags besides NONE.</param>
+        /// <param name="property">Property that is evaluated from 'expression'</param>
+        /// <param name="error">Error when evaluating 'expression'</param>
+        /// <returns>An HRESULT signifiyng if it sucessfuly obtainied a property or not from the expression.</returns>
+        private int GetIDebugPropertyFromExpression(string expression, IDebugStackFrame2 frame, enum_EVALFLAGS additionalEvalFlags, DAPEvalFlags additionalDapEvalFlags, out IDebugProperty2 property, out string error)
+        {
+            property = null;
+            error = string.Empty;
+
+            int hr;
+            IDebugExpressionContext2 expressionContext;
+            hr = frame.GetExpressionContext(out expressionContext);
+            if (hr >= 0)
+            {
+                IDebugExpression2 expressionObject;
+                uint errorIndex;
+                hr = expressionContext.ParseText(expression, enum_PARSEFLAGS.PARSE_EXPRESSION, Constants.ParseRadix, out expressionObject, out error, out errorIndex);
+                if (!string.IsNullOrEmpty(error))
+                {
+                    // TODO: Is this how errors should be returned?
```

Comment:
>// TODO: Is this how errors should be returned? [](start = 20, length = 47)

Looks correct to me

Reply:
I think this is copied from HandleEvaluateRequestAsync. you will want to use a different error code here so you know its coming through this path and not the previous.